File: leetcode/editor/cn/1455_检查单词是否为句中其他单词的前缀_test.py
import unittest


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        # given
        prefix_of_word = self.solution.isPrefixOfWord('i love eating burger',
                                                      'burge')
        # when
        self.assertEqual(4, prefix_of_word)


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        ans: int = -1
        words: list[str] = sentence.split(" ")
        for idx in range(len(words)):
            if words[idx].startswith(searchWord):
                ans = idx + 1
                break
        return ans


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()

File: leetcode/editor/cn/1455_检查单词是否为句中其他单词的前缀_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"strings"
	"testing"
)

func Test_givenNormal_when1455_thenSuccess(t *testing.T) {
	// when
	prefixOfWord := isPrefixOfWord("i love eating burger", "burg")
	// then
	assert.Equal(t, 4, prefixOfWord, "shouldEqual")
}

//leetcode submit region begin(Prohibit modification and deletion)
func isPrefixOfWord(sentence string, searchWord string) int {
	ans := -1
	words := strings.Split(sentence, " ")
	for idx, word := range words {
		if strings.HasPrefix(word, searchWord) {
			ans = idx + 1
			break
		}
	}
	return ans
}

//leetcode submit region end(Prohibit modification and deletion)
